tests/cspc/provisioning/provisioning_test.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	"github.com/openebs/cstor-operators/tests/pkg/cspc/cspcspecbuilder"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+/*
+This test file covers following test cases :
+
+1. Stripe pool provisioning with multiple disks and raid groups
+   ( includes write cache and data raid groups )
+2. Mirror pool provisioning with multiple disks and raid groups
+   (includes write cache and data raid groups )
+3. Raidz1 pool provisioning with multiple disks and raid groups
+   (includes write cache and data raid groups )
+4. Raidz2 pool provisioning with multiple disks and raid groups
+   (includes write cache and data raid groups )
+
+NOTE: The test cases adjusts depending on the number of nodes
+in the Kubernetes cluster.
+Meaning, if only 1 node is present then the test result expectations(output)
+are in accordance with what it should be with 1 node.
+
+if only 3 node is present then the test result expectations(output)
+are in accordance with what it should be with 3 node.
+
+Before starting the test suite, it should be specified whether it is
+a 3 node or 1 node test.
+
+Test suite only supports either a 1 node or 3 node test.
+
+*/
+
+var _ = Describe("CSPC Stripe", func() {
+	ProvisioningTest("stripe", 1)
+	ProvisioningTest("mirror", 2)
+	ProvisioningTest("raidz", 5)
+	ProvisioningTest("raidz2", 6)
+})
+
+func ProvisioningTest(poolType string, bdCount int) {
+	var cspc *cstor.CStorPoolCluster
+	Describe("Provisioning and cleanup of CSPC", func() {
+		Context("Creating cspc", func() {
+			Specify("no error should be returned", func() {
+				cspc = cspcspecbuilder.
+					NewCSPCSpecBuilder(cspcsuite.CSPCCache, cspcsuite.infra).
+					BuildCSPC(poolType, bdCount).
+					GetCSPCSpec()
+				_, err := cspcsuite.
+					client.
+					OpenEBSClientSet.
+					CstorV1().
+					CStorPoolClusters(cspc.Namespace).
+					Create(cspc)
+				Expect(err).To(BeNil())
+			})
+
+			Specify("desired count should be on cspc",
+				func() {
+					gotCount := cspcsuite.
+						client.
+						GetDesiredInstancesStatusOnCSPC(cspc.Name, cspc.Namespace, int32(cspcsuite.infra.NodeCount))
```

Comment:
Same as previous comment here NodeCount doesn't fit. Can we rename it as poolCount?

Reply:
NodeCount is intuitive from infra point. It is not a pool property. 
We can expect pool count to be equal to some other count -- e.g NodeCount, VolumeCount. 
This segregation allows viewing the relation and expectation.